### DISCUSSION TITLE
[@types/markdown-it]: Confusion importing 'markdown-it'

### DISCUSSION THREAD

**Discussion by Subilan** [DefinitelyTyped/DefinitelyTyped#67467] [Issues with a @types package]
After successfully installing the package `@types/markdown-it`, a direct ES6 `import` cannot be done.

<img width="926" alt="image" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/36447840/7dd554bb-a6fd-4f59-ae82-f24ad9217ecd">

For now the `compilerOptions` in `tsconfig.json` looks like this:
```json
"compilerOptions": {
        "noImplicitAny": true,
        "declaration": true,
        "outDir": "./dist/",
        "module": "ES6",
        "target": "ES5",
        "allowJs": true
}
```

So I turned `allowSyntheticDefaultImports` on and the importing issue was solved, however the imported `markdown-it` now has no type delcaration.

<img width="442" alt="image" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/36447840/e05639ec-b55f-4625-8435-425f1024942c">

I can't figure out why, so I turned to `node_modules/@types/markdown-it/index.d.ts` and found this 

<img width="1098" alt="image" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/36447840/0fbfc023-560c-4243-b23a-480cf3b2a6e4">

Truly if I set `moduleResolution` to `NodeNext` this problem can be partly solved, but apparently it also requires me to set `module` to `NodeNext`, according to VSCode. These steps made this approach seemingly a lucky hit rather than the right solution to me. If I change the first line of `index.d.ts`, the problem can be partly solved as well:

```diff
- import MarkdownIt = require("./lib");
+ import MarkdownIt = require("./lib/index");
```

By changing `./lib` to `./lib/index`, the importing target becomes more explicit. But now the imported is just the constructor class.

In all, I am just guessing. I wonder what is the right usage of the type declarations, and is there any document or note that can explain that?

[a comment by an automated account is not shown]

**Reply by peterblazejewicz** [reply]
the module export types as:
```ts
export = MarkdownIt;
```
(for imports), so please add:
```json
"esModuleInterop": true,
```
to your `tsconfig`:
https://www.typescriptlang.org/tsconfig#esModuleInterop
this should work with your setup
